python-stdlib/unittest/manifest.py

```diff
@@ -0,0 +1,3 @@
+metadata(version="0.9.0")
+
+module("unittest.py")
```

Comment:
I guess there's no way to specify that only `unittest_discover.py` requires `argparse`/`fnmatch`.

Maybe instead `unittest_discover.py` could be a separate module/package?

Reply:
I guess so... I don't really know how people use this package.

The CPython way is `python -m unittest discover` (or just `python -m unittest`). So if we moved it into `unix-ffi/unitest_discover` then you'd have to write `micropython -m unittest_discover`.

Reply:
How about wrapping the import in a try/catch so that normal usage keeps on working, and discovery is just unavailable unless argparse and fnmatch are available? It's a bit lying about dependencies of course..

Reply:
Yeah I'd prefer to not need to move it to a different package certainly! discover was split to the separate file with the intention of being able to install it or not depending on whether it needed to be used... 

I'm hoping this issue could be resolved with manifest options like https://github.com/micropython/micropython-lib/blob/f3cfc52ab076fc913dc6e266fb7370b418c8f542/micropython/bluetooth/aioble/manifest.py#L7

Hopefully can use that to modify dependencies too?

Reply:
> How about wrapping the import in a try/catch so that normal usage keeps on working, and discovery is just unavailable unless argparse and fnmatch are available? It's a bit lying about dependencies of course..

I put the import in the discover function so that normal functionality should work without the unittest_discover.py file... but perhaps an explicit try/catch with helpful user error would be better.

Reply:
> but perhaps an explicit try/catch with helpful user error would be better.

No I don't think that's necessary.  The code is good as-is.

The issue is really just about dependencies, that you'd need to explicitly install `argparse` and `fnmatch` on unix to use the `discover()` function.  Maybe fixing/improving that can be left until later.

Reply:
How does this look @jimmo - do the options need to be declared anywhere? https://github.com/micropython/micropython-lib/pull/530